### PR TITLE
Expanding unit test suite

### PR DESCRIPTION
**Pull request by andyfry01**
Hey @appleboy 

Here is a PR for #221. 

To summarize the changes: 
- I added testing for the component props and the component methods. 
- I added Enzyme and JSDOM to the test suite, which makes it possible to test aspects of the component that require a DOM (see [here](http://airbnb.io/enzyme/docs/guides/jsdom.html) for more info).
- Component prop testing checks whether the component renders props that are passed in and whether it renders default props successfully when no props are passed.
- Component method testing consists of checking basic method functionality (whether it updates state appropriately on the component, whether it calls other component methods successfully, etc.).
- Added a `test:watch` script to package.json so pre-commit doesn't get hung up because of the `--watch` flag on Jest. 

(also accidentally included [834f4ae](https://github.com/appleboy/react-recaptcha/commit/834f4ae7d664698c59aae22622873052814da09b), which was a commit for [this PR](https://github.com/appleboy/react-recaptcha/pull/220). Forgot to switch branches before committing my unit test commits)

Thank you! Let me know if you have any question. 
Andy

### CODE REVIEW

File: __tests__/recaptcha-test.js [hunk 1]
```diff
@@ -1,13 +1,182 @@
...
+	sitekey: 'testSitekey',
+	theme: 'testTheme',
+	type: 'testType',
+	verifyCallbackName: 'testVerifyCallbackName',
+	expiredCallbackName: 'testExpiredCallbackName',
+	size: 'testSize',
+	tabindex: 'testTabindex',
+	hl: 'testHl',
+	badge: 'testBadge'
+};
```
Comment (appleboy) [inline __tests__/recaptcha-test.js]: coding style? Is indent two space?

Comment (andyfry01) [reply __tests__/recaptcha-test.js]: Looks like indenting is tabs right now, should I change it to two spaces?

Comment (andyfry01) [reply __tests__/recaptcha-test.js]: See [e7ed11d](https://github.com/appleboy/react-recaptcha/pull/223/commits/e7ed11dcb8a589ca93a8d6a7f5a65246bd2cc8e6)

**Comment by appleboy** [comment]
maybe we can drop the node 4 and 5 support. 

https://travis-ci.org/appleboy/react-recaptcha/builds/334740273

**Comment by andyfry01** [comment]
Node v6 was released in spring 2016, so dropping support for v4/5 seems ok. It looks like Travis is failing at the yarn build step, so maybe transitioning from yarn to npm could be an alternative to dropping v4/5 support. Npm has package-lock files now, and that was the primary benefit of yarn vs older versions of npm.

**Comment by appleboy** [comment]
@andyfry01 Could you help to switch `yarn` to `npm`? Thanks.

**Comment by andyfry01** [comment]
Sure thing, I'll do a little research and let you know.

**Comment by andyfry01** [comment]
Hey @appleboy, after looking into this more closely, it seems like the issue goes deeper than `yarn`. It looks like support for JSDOM and Jest in Node < v6 is limited (see [this thread](https://github.com/facebook/jest/issues/5211)). I tried to implement [this workaround](https://github.com/facebook/jest/issues/5211#issuecomment-354956892), but it's a bit above my head to be honest and I'm having a hard time getting it to work.

I could spend some more time on this, or if you don't mind dropping v4/5 support then that can be an option.

**Comment by appleboy** [comment]
@andyfry01 Maybe we can drop v4/5 support and update the message in README. OK?

**Comment by andyfry01** [comment]
Hey @appleboy, updated the readme and dropped Node v4/5 from travis.yml.

**Pull request merged**